web3.js/src/publickey.ts

```diff
@@ -147,10 +147,10 @@ export class PublicKey extends Struct {
     seed: string,
     programId: PublicKey,
   ): Promise<PublicKey> {
-    const buffer = Buffer.concat([
-      fromPublicKey.toBuffer(),
-      Buffer.from(seed),
-      programId.toBuffer(),
+    const buffer = Uint8Array.from([
```

Comment:
We can use `new Uint8Array(length)` and `set()` to build up this array with fewer allocations and less for the garbage collector to clean up.